### ISSUE TITLE
Ability to use Pipeline Expressions within ECS task definition.json

### ISSUE THREAD

**Issue by away168** [spinnaker/spinnaker#6113]
### Issue Summary:
The desired behavior is the ability to use pipeline expressions (SpEL) inside the task definition artifact.  Currently, Pipelines expressions within Task Definitions artifacts are not "evaluated" by Spinnaker, which results in an AWS error of improper value. 

### Cloud Provider(s):
AWS

### Environment:
Spinnaker 1.21+, ECS

### Feature Area:
Orca, Clouddriver

### Description:
ECS allows for defining the task using a task definition json from an artifact source like GitHub.  
The desire is to embed pipeline expressions within the task definition json so that it would be evaluated by Spinnaker before being submitted to AWS.

Example usage:  Dynamically configuring the arn of the secret key in this example: 
https://docs.aws.amazon.com/AmazonECS/latest/userguide/specifying-sensitive-data-tutorial.html

```
{
    "executionRoleArn": "arn:aws:iam::aws_account_id:role/ecsTaskExecutionRole",
    "containerDefinitions": [
        {
            "entryPoint": [
                "sh",
                "-c"
            ],
            "portMappings": [
                {
                    "hostPort": 80,
                    "protocol": "tcp",
                    "containerPort": 80
                }
            ],
            "command": [
                "/bin/sh -c \"echo '<html> <head> <title>Amazon ECS Sample App</title> <style>body {margin-top: 40px; background-color: #333;} </style> </head><body> <div style=color:white;text-align:center> <h1>Amazon ECS Sample App</h1> <h2>Congratulations!</h2> <p>Your application is now running on a container in Amazon ECS.</p> </div></body></html>' >  /usr/local/apache2/htdocs/index.html && httpd-foreground\""
            ],
            "cpu": 10,
            "secrets": [
                {
                    "valueFrom": "${parameters.arnsecretusername}",
                    "name": "username_value"
                }
            ],
            "memory": 300,
            "image": "httpd:2.4",
            "essential": true,
            "name": "ecs-secrets-container"
        }
    ],
    "family": "ecs-secrets-tutorial"
}
```

**Comment by allisaurus** [comment]
Adding a note that this feature could be one way of accomplishing https://github.com/spinnaker/spinnaker/issues/4622

**Comment by akshayabd** [comment]
@away168 Do artifacts in general support the ability to use SpEL? Wondering if this is more relevant for the platform SIG to look at?

**Comment by away168** [comment]
yes, i know artifacts used for deploy (manifest) still go through SpEL processing. 
also bake (manifest) - for helm - has that capability too.

**Comment by away168** [comment]
> @away168 Do artifacts in general support the ability to use SpEL? Wondering if this is more relevant for the platform SIG to look at?

yes, i know artifacts used for deploy (manifest) still go through SpEL processing. 
also bake (manifest) - for helm - has that capability too.

**Comment by jasonmcintosh** [comment]
Given previous PRs, like the proposal:
https://github.com/spinnaker/spinnaker/issues/6374#issuecomment-5336641358
I think this is done and completed, so closing.
